cmd/clusterctl/client/cluster/objectgraph_test.go

```diff
@@ -1378,6 +1379,67 @@ func getObjectGraphWithObjs(objs []client.Object) *objectGraph {
 	return newObjectGraph(fromProxy, inventory)
 }
 
+// Similar to the objectgraph Discvoery method, but populates objects using the restoreAddObj method
+// This then can be used in testing to accurately build an object graph as if from file
+func restoreDiscovery(o *objectGraph, namespace string) error {
```

Comment:
Good call. This didn't make sense. I ended up removing this and added code in the tests to populate a temporary directory with the required object files (much like it would actually work) instead of relying on a bespoke `discovery` method that adds objects